tests/annotator/annotator.cpp

```diff
@@ -1446,6 +1468,27 @@ TEST(Annotator, automaticIdAllItemsWrongModel)
     EXPECT_EQ("", annotator->assignId(itemVariable));
 }
 
+TEST(Annotator, issueIdWithModelThatWasDeleted)
+{
+    libcellml::ModelPtr model;
+    auto parser = libcellml::Parser::create();
+    model = parser->parseModel(modelStringUniqueIds);
+    auto annotator = libcellml::Annotator::create();
+
+    annotator->setModel(model);
+
+    auto component = model->component(0);
+    libcellml::AnyItem itemComponent = std::make_pair(libcellml::CellMLElement::COMPONENT, model->component(0));
+    EXPECT_EQ("component_1", component->id());
+
+    EXPECT_TRUE(annotator->hasModel());
+    model.reset();
+    EXPECT_FALSE(annotator->hasModel());
+
+    EXPECT_EQ("b4da55", annotator->assignId(itemComponent));
```

Comment:
What happens when this component is added to a model? or another component without a parent model is passed to the annotator?  I think if the item passed in for id isn't a part of the stored model (or there is no stored model!) it should return an error instead of assigning an id which can't be said to be unique as it has no context.

Reply:
Well if the component has no model then it is unique for the given context.  If you go and add it to a model the context has changed, nothing the library can do about that.

I don't see why it should return an error.  Someone has given me a component and they want an id assigned to it.  I can do that so I do, no problem.

Reply:
The only reason to assign an id using the Annotator is to assign something unique.  As soon as there's no model, it can't make it unique.  So it's kinda pointless.

Reply:
I will change the documentation so that it's clear that the model must be set first.